apiserver/pkg/model/converter.go

```diff
@@ -165,3 +165,113 @@ func FromCrdToApiJob(job *v1alpha1.RayJob) (pbJob *api.RayJob) {
 	}
 	return pbJob
 }
+
+func FromCrdToApiServices(services []*v1alpha1.RayService, serviceEventsMap map[string][]v1.Event) []*api.RayService {
+	apiServices := make([]*api.RayService, 0)
+	for _, service := range services {
+		apiServices = append(apiServices, FromCrdToApiService(service, serviceEventsMap[service.Name]))
+	}
+	return apiServices
+}
+
+func FromCrdToApiService(service *v1alpha1.RayService, events []v1.Event) *api.RayService {
+	defer func() {
+		err := recover()
```

Comment:
will we have any panic issue below?

Reply:
here is a check for the unexpected situation, I just make it aligned with the ray jobs https://github.com/ray-project/kuberay/blob/fdee883c3de7fa95fca6659bd200b98f2cc82b02/apiserver/pkg/model/converter.go#L130-L136

Reply:
sounds good